pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -550,14 +550,15 @@ public final synchronized void readEntriesComplete(List<Entry> entries, Object c
         if (serviceConfig.isDispatcherDispatchMessagesInSubscriptionThread()) {
             // setting sendInProgress here, because sendMessagesToConsumers will be executed
             // in a separate thread, and we want to prevent more reads
+            havePendingRead = true;
```

Comment:
Just mark here, we set `havePendingRead` here has a risk if the task in `dispatchMessagesThread` executor got some problem. the `havePendingRead` will never be set to false.

But I'm not quite sure if the problem exists because this executor is an unbounded queue.

Reply:
That is a good point. We could handle the `RejectedExecutionException` if we think it will happen. As you said though, the queue is unbounded, so I don't think it is a case we _have_ to handle.